tests/test_writer.py

```diff
@@ -0,0 +1,59 @@
+import os
+import fcntl
+from multiprocessing import Process, Queue
+from datetime import datetime
+import pytest
+from warcprox.mitmproxy import ProxyingRecorder
+from warcprox.warcproxy import RecordedUrl
+from warcprox.writer import WarcWriter
+from warcprox import Options
+
+recorder = ProxyingRecorder(None, None, 'sha1', url='http://example.com')
+
+recorded_url = RecordedUrl(url='http://example.com', content_type='text/plain',
+                           status=200, client_ip='5.5.5.5',
+                           request_data=b'abc',
+                           response_recorder=recorder,
+                           remote_ip='6.6.6.6',
+                           timestamp=datetime.utcnow())
+
+
+def lock_file(queue, filename):
+    """Try to lock file and return 1 if successful, else return 0.
+    It is necessary to run this method in a different process to test locking.
+    """
+    try:
+        fi = open(filename, 'ab')
+        fcntl.lockf(fi, fcntl.LOCK_EX | fcntl.LOCK_NB)
+        fi.close()
+        queue.put('1')
```

Comment:
There's no reason these have to be strings that look like numbers, `'1'`, they could be anything, e.g. `1`, `True`, or even `'OBTAINED LOCK'` / `'FAILED TO OBTAIN LOCK'` -- I might go with that, it's pretty clear

Reply:
Done